# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1707944763 -3600
# Parent  93b116012ef3f4740367701254a42fe9b785444e
Bug 1830439, bug 612800 - QF tag bar doesn't refresh when other quick filter conditions change.

Code lifted from alta88's 'Total Quick Filter' add-on.

diff --git a/mail/modules/QuickFilterManager.jsm b/mail/modules/QuickFilterManager.jsm
--- a/mail/modules/QuickFilterManager.jsm
+++ b/mail/modules/QuickFilterManager.jsm
@@ -734,19 +734,20 @@ var TagFacetingFilter = {
       term.value = value;
       term.op = aFilterValue
         ? Ci.nsMsgSearchOp.IsntEmpty
         : Ci.nsMsgSearchOp.IsEmpty;
       term.booleanAnd = !showUntagged;
       aTerms.push(term);
 
       // we need to perform faceting if the value is literally true.
-      if (aFilterValue === true) {
-        return this;
-      }
+      // NOTE: Fix Bug 612800; always return |this| listener if tags are active,
+      // to reflect tag buttons per filtered messages results, sort of obvious.
+      // if (aFilterValue === true)
+      return this;
     } else {
       let firstIncludeClause = true,
         firstExcludeClause = true;
       let lastIncludeTerm = null;
       term = null;
 
       let excludeTerms = [];
 
